File: AlternativeQL/src/org/ql/ast/literal/StringLiteral.java
package org.ql.ast.literal;

import org.ql.ast.Node;

public class StringLiteral implements Node {
    private String stringLiteral;

    public StringLiteral(String stringLiteral) {
        this.stringLiteral = stringLiteral;
    }

    public String toString() {
        return stringLiteral;
    }
}

File: OffByOne/src/OffByOne.LanguageCore/Ast/Literals/StringLiteral.cs
namespace OffByOne.LanguageCore.Ast.Literals
{
    using OffByOne.LanguageCore.Ast.Literals.Base;
    using OffByOne.LanguageCore.Visitors.Contracts;

    public class StringLiteral : Literal
    {
        public StringLiteral(string value)
        {
            this.Value = value.Trim('"');
        }

        public string Value { get; private set; }

        public override TResult Accept<TResult>(ILiteralVisitor<TResult> visitor)
        {
            return visitor.Visit(this);
        }
    }
}
